lib/learn_open/services/git_ssh_connector.rb

```diff
@@ -0,0 +1,40 @@
+module LearnOpen
+  class GitSSHConnector
+    attr_reader :ssh_connection, :environment
+
+    GIT_SSH_USER = 'git'
+
+    def self.call(git_server:, environment:)
+      self.new(git_server: git_server, environment: environment).call
+    end
+
+    def initialize(git_server:, environment:)
+      @ssh_connection = LearnOpen.ssh_adapter.new(user: GIT_SSH_USER, hostname: git_server)
+      @environment = environment
+    end
+
+    def call
+      if managed_environment? && ssh_unauthenticated?
+        upload_ssh_keys!
+      end
+
+      ssh_authenticated?
```

Comment:
I could cut an SSH check over the network here... this felt a little less brittle though? Granted the SSH connection test doesn't have any retry logic built yet.